frontend/src/components/pages/TranslationPage.tsx

```diff
@@ -101,35 +73,22 @@ const TranslationPage = () => {
     }
   };
 
-  const [updateTranslation] = useMutation<{
-    updateStoryTranslationContentById: UpdateTranslationResponse;
-  }>(UPDATE_TRANSLATION);
   const onChangeTranslationContent = async (
-    storyTranslationContentId: number,
     newContent: string,
+    lineIndex: number,
   ) => {
-    try {
-      const result = await updateTranslation({
-        variables: {
-          storyTranslationContentId,
-          newTranslationContent: newContent,
-        },
-      });
+    const updatedContentArray = [...translatedStoryLines];
+    const index = arrayIndex(lineIndex);
+    updatedContentArray[index].translatedContent = newContent;
 
-      const lineIndex =
-        result.data?.updateStoryTranslationContentById.story.lineIndex;
-      if (lineIndex !== undefined) {
-        const updatedContentArray = [...translatedStoryLines];
-        const index = arrayIndex(lineIndex);
-        updatedContentArray[index].translatedContent = newContent;
+    setTranslatedStoryLines(updatedContentArray);
+    setChangedStoryLines(
+      changedStoryLines.set(lineIndex, updatedContentArray[index]),
+    );
+  };
 
-        setTranslatedStoryLines(updatedContentArray);
-      } else {
-        handleError("Unable to save translation.");
-      }
-    } catch (err) {
-      handleError(err ?? "Error occurred, please try again.");
-    }
+  const clearUnsavedChangesMap = () => {
+    setChangedStoryLines(new Map());
```

Comment:
There may be an unnecessary render triggered by this?

Reply:
Hmmm I think this is inevitable. I think the only way to reduce rerenders is to make `changedStoryLines` non state, but in the current code on this pr, `changedStoryLines` needs to be in react state for the `Autosave` to debounce properly.